src/chart_process/bmson_processor.rs

```diff
@@ -393,7 +393,7 @@ impl ChartProcessor for BmsonProcessor {
             // Calculate display ratio: (event_y - current_y) / visible_window_y * scroll_factor
             // Note: scroll can be non-zero positive or negative values
             let display_ratio_value = if visible_window_y > YCoordinate::zero() {
-                (&(event_y - current_y) / &visible_window_y).value() * scroll_factor
+                Decimal::from(&(event_y - current_y) / &visible_window_y) * scroll_factor.clone()
```

Comment:
```suggestion
                &Decimal::from(&(event_y - current_y) / &visible_window_y) * scroll_factor
```
This `clone` can be reduced by aligning ref.